Description: Copy over GTK 3 settings from XDG_CONFIG_DIRS
 This is a workaround for GTK 3 settings not being applied by default from XDG.
 Ideally, all desktop environments would do this, and it would be fixed in GTK itself.
Author: Simon Quigley <tsimonq2@ubuntu.com>
Origin: vendor
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gtk+3.0/+bug/2047705
Forwarded: no
Last-Update: 2023-12-29
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/startlxqt.in
+++ b/startlxqt.in
@@ -32,6 +32,21 @@ else
         fi
     done
 fi
+# Handle GTK 3 configuration files
+echo "$XDG_CONFIG_DIRS" | tr ':' '\n' | while read -r dir; do
+    dir_path="$dir/gtk-3.0/"
+    if [ -d "$dir_path" ]; then
+        mkdir -p "$HOME/.config/gtk-3.0"
+        for file in "$dir_path"*; do
+            if [ -f "$file" ]; then
+                target_file="$HOME/.config/gtk-3.0/$(basename "$file")"
+                if [ ! -f "$target_file" ]; then
+                    cp "$file" "$target_file"
+                fi
+            fi
+        done
+    fi
+done
 
 if [ -z "$XDG_CACHE_HOME" ]; then
     export XDG_CACHE_HOME="$HOME/.cache"
